chore(readme): improve structure and content (#246)
ngrok/kubernetes-ingress-controller: README.md

> <p align="center">

Reviewer: I do prefer the way it looks centered, but its not a huge deal. It would be great to get some ngrok + language logos to have consistent headers and look/feels for our open source repos

  </a>
</p>

# ngrok Kubernetes Ingress Controller

Leverage [ngrok](https://ngrok.com/) for your [Kubernetes Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/). Instantly add load balancing, authentication, and observability to your services via ngrok Cloud Edge modules using Custom Resource Definitions (CRDs) and Kubernetes-native tooling.

[Installation](#installation) | [Getting Started](https://ngrok.com/docs/using-ngrok-with/k8s/) | [Documentation](#documentation) | [Developer Guide](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/docs/developer-guide/README.md) | [Known Issues](#known-issues)

## Installation

### Helm

> > **Note** We recommend using the Helm chart to install the controller for a better experience through changes.

Author: ```suggestion
> **Note** We recommend using the Helm chart to install the controller for a better experience for upgrades.
```

```

For a more in-depth installation guide follow our step-by-step [Getting Started](https://ngrok.com/docs/using-ngrok-with/k8s/) guide.

## Documentation

The full documentation for the ngrok Ingress Controller can be found [under the docs directory](./docs/README.md). Pull requests for corrections and additions are always welcomed.

###  Guides and Tutorials
- [Deployment Guide](./docs/deployment-guide/README.md): for installing the controller for the first time
- [User Guide](./docs/user-guide/README.md): for an in depth view of the ngrok ingress configuration options and primitives
- [Examples](./docs/examples/README.md): for examples of how to configure ingress in different scenarios (e.g. Hello World, Consul, OAuth, etc.)
- [Developer Guide](./docs/developer-guide/README.md): for those interested in contributing to the project

## Known Issues

> **Note** 
>
> This project is currently in beta as we continue testing and receiving feedback. The functionality and CRD contracts may change. It is currently used internally at ngrok for providing ingress to some of our production workloads. 

> 1. The ngrok Ingress Controller requires a ngrok Pro or Enterprise subscription.

Reviewer 2: ```suggestion
```

found one more instance to remove!

  </a>
</p>

# ngrok Kubernetes Ingress Controller

> > As of today, the ngrok Ingress Controller only works with an ngrok Pro subscription or above. If you would like to test out the ngrok Ingress Controller with a free ngrok account, please [reach out to us via slack](https://ngrok.com/slack) and we will be happy to help.

Reviewer 2: ```suggestion

```

Leverage [ngrok](https://ngrok.com/) for your [Kubernetes Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/). Instantly add load balancing, authentication, and observability to your services via ngrok Cloud Edge modules using Custom Resource Definitions (CRDs) and Kubernetes-native tooling.

[Installation](#installation) | [Getting Started](https://ngrok.com/docs/using-ngrok-with/k8s/) | [Documentation](#documentation) | [Developer Guide](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/docs/developer-guide/README.md) | [Known Issues](#known-issues)

## Installation

### Helm

> > **Note** We recommend using the Helm chart to install the controller for a better experience through changes.

Reviewer: > through changes

nit: maybe a better experience for upgrades?

Add the ngrok Ingress Controller Helm chart:

```sh
helm repo add ngrok https://ngrok.github.io/kubernetes-ingress-controller
```

Then, install the latest version (setting the appropriate values for your environment):

```sh
export NAMESPACE=[YOUR_K8S_NAMESPACE]
export NGROK_AUTHTOKEN=[AUTHTOKEN]
export NGROK_API_KEY=[API_KEY]

helm install ngrok/kubernetes-ingress-controller \

>   --generate-name \

Reviewer: I think this is a flag to emulate the old way of generating a name in helm2 and isn't very standard to see for most helm examples https://stackoverflow.com/questions/57961162/helm-install-unknown-flag-name

if this is an attempt resolve https://github.com/ngrok/kubernetes-ingress-controller/issues/179 we can go with for now and understand the best option in that issue

Author: Yes, as a part of the discussion that came out of #179 was that by using the name would append as the release name instead of when using generate-name it gets omitted and makes the names shorter. Would love to have that discussion with you at some time as well!

But, I agree, let's revert this and push it through #179 instead!

  --namespace $NAMESPACE \
  --create-namespace \
  --set credentials.apiKey=$NGROK_API_KEY \
  --set credentials.authtoken=$NGROK_AUTHTOKEN
```

> **Note** The values for `NGROK_API_KEY` and `NGROK_AUTHTOKEN` can be found in your [ngrok dashboard](https://dashboard.ngrok.com/get-started/setup) and are used by your ingress controller to authenticate with ngrok for configuring and running your network ingress traffic at the edge.

For a more in-depth installation guide follow our step-by-step [Getting Started](https://ngrok.com/docs/using-ngrok-with/k8s/) guide.

### YAML Manifests

Apply the [sample combined manifest](manifest-bundle.yaml) from our repo:

```sh
kubectl apply -n ngrok-ingress-controller \
  -f https://raw.githubusercontent.com/ngrok/kubernetes-ingress-controller/main/manifest-bundle.yaml
```

For a more in-depth installation guide follow our step-by-step [Getting Started](https://ngrok.com/docs/using-ngrok-with/k8s/) guide.

## Documentation

> You can find the full documentation for the ngrok Ingress Controller can be found [under the docs directory](./docs/README.md). Pull requests for corrections and additions are always welcomed.

Reviewer: ```suggestion
The full documentation for the ngrok Ingress Controller can be found [under the docs directory](./docs/README.md). Pull requests for corrections and additions are always welcomed.
```
